### PR TITLE
Use patch to update shoot status

### LINKED ISSUE #5825: Use `Patch` instead of `Update` when setting `Shoot` status to `Aborted`
**How to categorize this issue?**

/area usability
/kind enhancement

**What would you like to be added**:
To prevent `gardenlet` from exiting, we should use `Patch` instead of `Update` when updating the `.status.lastOperation.state` to `Aborted` on start-up: https://github.com/gardener/gardener/blob/61e391ecb37013c8fa0637136bb03ba9f4ddac91/pkg/gardenlet/controller/shoot/shoot.go#L192-L199
We only care about updating the status properly and don't need to operate on the most recent version of the object in this case.

**Why is this needed**:
Prevent unnecessary restarts.

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area usability
/kind enhancement

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #5825

**Special notes for your reviewer**:
I am not sure if something else is required. I tried to check similar usages of update in gardenlet but found only this.

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/gardenlet/controller/shoot/shoot.go [hunk 1]
```diff
@@ -192,8 +192,9 @@ func (c *Controller) Run(ctx context.Context, shootWorkers, shootCareWorkers, sh
 		// Check if the status indicates that an operation is processing and mark it as "aborted".
 		if shoot.Status.LastOperation != nil && shoot.Status.LastOperation.State == gardencorev1beta1.LastOperationStateProcessing {
 			newShoot := shoot.DeepCopy()
+			patch := client.MergeFrom(newShoot)
```
Comment (rfranzke) [inline pkg/gardenlet/controller/shoot/shoot.go]: Shouldn't this be

```suggestion
			patch := client.MergeFrom(newShoot.DeepCopy())
```

?

Comment (acumino) [reply pkg/gardenlet/controller/shoot/shoot.go]: `newShoot` is already Deepcopy.

**Review by krgostev — COMMENTED** [review]
/lgtm

**Review by timebertt — APPROVED** [review]
/lgtm

**Pull request merged**